mewarpx/mewarpx/emission.py

```diff
@@ -1179,20 +1180,158 @@ def get_normals(self, x, y, z):
+                used to obtain work function and z coordinate/direction.
+            T (float): Temperature in Kelvin for the emitter; determines
+                velocities. If not specified the temperature of the conductor
+                will be used.
+            x (float): Position of the emitter along the x axis. Default
+                conductor.x if it exists otherwise conductor.xmin/max depending
+                on xdir. If none of those attributes exist an error will be
+                raised if x is not specified.
+            ymin (float): Minimum position of the rectangular emitter along y.
+                Default conductor.ymin if it exists otherwise mwxrun.ymin.
+            ymax (float): Maximum position of the rectangular emitter along y.
+                Default conductor.ymax if it exists otherwise mwxrun.ymax.
+            zmin (float): Minimum position of the rectangular emitter along z.
+                Default conductor.zmin if it exists otherwise mwxrun.zmin.
+            zmax (float): Maximum position of the rectangular emitter along z.
+                Default conductor.zmax if it exists otherwise mwxrun.zmax.
+            transverse_fac (float): Scale the transverse energy distribution by
+                this factor. Default 1. See
+                :func:`mewarpx.utils_store.util.get_velocities` for details.
+            xdir (int): 1 to emit in +x, -1 to emit in -x.
+            kwargs (dict): Any other keyword arguments supported by the parent
+                Emitter constructor (such as "emission_type").
+        """
+        # Default initialization
+        super(XPlaneEmitter, self).__init__(T=T, conductor=conductor, **kwargs)
+
+        self.transverse_fac = transverse_fac
+        self.xdir = int(round(xdir))
+        if self.xdir not in [-1, 1]:
+            raise ValueError("xdir must be +1 or -1 for x-plane emitters.")
+
+        self.x = x
+        if self.x is None:
+            try:
+                self.x = conductor.x
+            except AttributeError:
+                try:
+                    if self.xdir == 1:
+                        attr = 'xmax'
+                    else:
+                        attr = 'xmin'
+                    self.x = getattr(conductor, attr)
+                except AttributeError:
+                    raise AttributeError(
+                        'x must be specified for x-plane emitter if the '
+                        f'attached conductor does not specify x or {attr}'
+                    )
+
+        # Determine bounds
+        # Will be 4 element array [ymin, ymax, zmin, zmax]
+        self.bounds = []
+        for bound in ['ymin', 'ymax', 'zmin', 'zmax']:
+            default = (
+                getattr(mwxrun, bound) if not hasattr(conductor, bound) else
+                getattr(conductor, bound)
+            )
+            bound = locals()[bound]
+            if bound is None:
+                bound = default
+            self.bounds.append(bound)
```

Comment:
Gah, even though I technically overwrote bound after it was done being used I didn't mean to. How about
```suggestion
        for boundstr in ['ymin', 'ymax', 'zmin', 'zmax']:
            bound = locals()[boundstr]
            if bound is None:
                bound = (
                    getattr(mwxrun, boundstr) if not hasattr(conductor, boundstr) else
                    getattr(conductor, boundstr)
                )
            self.bounds.append(bound)
```